astroquery/simbad/core.py

```diff
@@ -936,6 +1006,243 @@ def query_objectids_async(self, object_name, *, cache=True,
 
         return response
 
+    def tables(self, get_adql=False):
+        """The names and descriptions of the tables in SIMBAD.
+
+        Parameters
+        ----------
+        get_adql : bool, optional
+            Returns the ADQL string instead of querying SIMBAD.
+
+        Returns
+        -------
+        `~astropy.table.table.Table`
+        """
+        query = ("SELECT table_name, description"
+                 " FROM TAP_SCHEMA.tables"
+                 " WHERE schema_name = 'public'")
+        if get_adql:
+            return query
+        return self.query_tap(query)
+
+    def columns(self, *tables: str, get_adql=False):
+        """
+        Get the list of SIMBAD columns.
+
+        Add tables names to restrict to some tables. Call the function without
+        any parameter to get all columns names.
+
+        Parameters
+        ----------
+        tables : str, optional
+            Add tables names as strings to restrict to these tables columns.
+        get_adql : bool, optional
+            Returns the ADQL string instead of querying SIMBAD.
+
+        Examples
+        --------
+        >>> from astroquery.simbad import Simbad
+        >>> Simbad.columns("ids", "ident") # doctest: +REMOTE_DATA
+        <Table length=4>
+        table_name column_name datatype ...  unit    ucd
+          object      object    object  ... object  object
+        ---------- ----------- -------- ... ------ -------
+             ident          id  VARCHAR ...        meta.id
+             ident      oidref   BIGINT ...
+               ids         ids     CLOB ...        meta.id
+               ids      oidref   BIGINT ...
+        """
+        query = ("SELECT table_name, column_name, datatype, description, unit, ucd"
+                 " FROM TAP_SCHEMA.columns"
+                 " WHERE table_name NOT LIKE 'TAP_SCHEMA.%'")
+        if len(tables) == 1:
+            query += f" AND table_name = '{tables[0]}'"
+        elif len(tables) > 1:
+            query += f" AND table_name IN ({str(tables)[1:-1]})"
+        query += " ORDER BY table_name, principal DESC, column_name"
+        if get_adql:
+            return query
+        return self.query_tap(query)
+
+    def find_columns_by_keyword(self, keyword: str, get_adql=False):
```

Comment:
maybe this can be merged into the previous method? by adding the optional `keyword` kwarg to it?

Reply:
Ah yes, good idea. This is implemented in the last commit from today :)